src/diffusers/image_processor.py

```diff
@@ -882,3 +883,56 @@ def preprocess(
             depth = self.binarize(depth)
 
         return rgb, depth
+
+
+class IPAdapterMaskProcessor(VaeImageProcessor):
+    """
+    Image processor for IP Adapter image masks.
+
+    """
+    def __init__(self):
+        super().__init__(do_normalize=False, do_binarize=True, do_convert_grayscale=True)
+
+    def process(self, images: Union[List[PIL.Image.Image], PIL.Image.Image]) -> torch.FloatTensor:
+        """
+        Convert a PIL.Image.Image or a list of PIL.Image.Image to a torch.FloatTensor
+        """
+        if not isinstance(images, list):
+            images = [images]
+        images = self.preprocess(images)
+        return images
+
+    @staticmethod
+    def downsample(mask: torch.FloatTensor, batch_size: int, seq_length: int, value_embed_dim: int):
```

Comment:
It's uncommon to see `seq_length` as a parameter for downsampling at least in the vision domain. Could we aim for a better and more well-understood argument name here?